classes/hydraulic_suitability_scenario/vulnerability/Vulnerability.py

```diff
@@ -0,0 +1,68 @@
+import os
+import glob
+from utils.helpers import read_csv_to_arrays
+from utils.constants import CEND, CRED
+
+from calculations.hydraulic_suitability_scenario.calculate_vulnerability import calculate_vulnerability
+
+
+class Vulnerability:
+    senario_files = 'files_output/hydraulic_suitability_scenario/*senario.csv'
+    percentile_files = 'files_output/hydraulic_suitability_scenario/*opt_percentile.csv'
+    col_num = {
+        'min': 0,
+        '10th': 1,
+        '25th': 2,
+        '50th': 3,
+        '75th': 4,
+        '90th': 5,
+        'max': 6
+    }
+    """Calculate Vulnerability
+
+    """
+
+    def __init__(self, limits, inputs):
+        self.limits = limits
+        self.inputs = inputs
+        self.vulnerability = [['File Name', 'Function', 'Vulnerability']]
+
+        self.num_of_func = len(self.limits)
+
+        self.files = ({'scenario': glob.glob(
+            self.senario_files), 'percentile': glob.glob(self.percentile_files)})
+
+        if not len(self.files):
+            print(
+                CRED + "Looks like you have not uploaded any files into input folders..."+CEND)
+            return
+
+        self.calc_vulnerability()
+
+    def read_scenario_and_opt_percentile(self, scenario, percentile):
```

Comment:
we should just be using the data already saved in memory instead of reading more of the same data into memory. op_dataset should have all the data you need.